### ISSUE TITLE
No output on commands which print something

### ISSUE THREAD

**Issue by spencerperley** [jonarrien/telescope-cmdline.nvim#11]
Not sure if I did not configure right or if this is not the intended functionality but when I enter a command like : map {some character} it does not show the output. The prompt to press enter to continue is given but not the actual output.

**Comment by jonarrien** [comment]
Yes, it should work. However, this plugin uses `vim.api.nvim_exec2()` and it was introduced in neovim version`0.8.0`. Are you using an older version?

If you are using an older neovim version and would like to make it work, I guess it's a matter of adding a condition [here](https://github.com/jonarrien/telescope-cmdline.nvim/blob/51ebf3e585a660a431cab4ed7352c608350b0633/lua/cmdline/actions.lua#L51) and use `vim.api.nvim_exec()` for older versions.

**Comment by jonarrien** [comment]
Added a condition to check for nvim version 0.8.0. I'm currently on 0.10.0-dev and can't reproduce your issue. Would you mind updating the plugin and giving it a try?

**Comment by spencerperley** [comment]
I just reinstalled and the error persisted. Im on version 0.9.5 nvim and check heath is all good. I reinstalled things by just adding the packer use statement you provided and running the command :silent Telescope cmdline and then running map s ( a command which prints two mappings with the normal stock pressing : and running the command) and it did not print.

**Comment by jonarrien** [comment]
I've tested in neovim 0.9.5 using `lazy` package manager and it is working fine. But if I use `packer`, you are right, is not showing any output... 😨

As alternative, you can enable `output_pane` in the extension settings. I've tested and it's working fine with `packer`.  This setting renders the output in split pane. 
* CONS: You need to close the pane afterwards (like any buffer)
* PROS: You can select and yank the output

```lua
local telescope = require("telescope")
telescope.setup({
  extensions = {
    cmdline = {
      output_pane = {
        enabled = true
      }
    },
  }
})

telescope.load_extension("cmdline")
```

> NOTE: I don't use packer anymore, but any pull request is welcome. Hope the output pane helps in the meantime.

**Comment by juliangoetze** [comment]
@jonarrien First of all, **I love the plugin**, it has been the missing piece of my puzzle (I've needed vim-normal-mode in the cmd-line for a long time).

But I do have the same issue.
I am using `vim-plug` and `NVIM v0.9.5`.

As I run commands that should yield an output, I can observe that the **output is being shown for fractions of a second and disappear** right away.

And with the suggested fix

```lua
telescope.setup({
  extensions = {
    cmdline = {
      output_pane = {
        enabled = true
      }
    },
  }
})
```

I observe something strange: 

- Commands that yield an output with focus on the new window like `:messages` work as expected, opening a new window.

- For commands that **yield output without focus** like `:echo 1`, the bug remains.

Do you have any workaround in mind? I would be sad to not use the plugin because of this...

**Comment by juliangoetze** [comment]
@jonarrien I investigated the error a bit further.

My discovery was, that in my case somehow the **notification-output of `Telescope` (that displays the actual vim-mode) and the output of the called `cmd` competed** against each other.
As a result, the `cmd`-output disappeared right away.

As a workaround, I wrapped the `vim.api.nvim_exec2(...)`-call into a `vim.defer_fn(function()` to **call it asynchronously** after 1ms. That fixed all the output-issues for me except for one:

Calling `:messages` from `Telescope cmdline` does never yield an output, when called via 

```lua 
vim.api.nvim_exec2('messages', { output= false }) 
```

although inspecting the output via

```lua 
local data = vim.api.nvim_exec2('messages', { output= true }) 
vim.notify(data.output)
```

does always yield the output correctly.

But handling it that way would **lead to exponential `messages` growth**, because the `notify`-message gets fed back right into the `messages` itself.

I can't explain the `:messages`-bug, but the fix for the other outputs works fine. Do you have anything in mind that could be the cause of the `:messages`-problem? And do the `:messages` work well in your environment?

**Comment by jonarrien** [comment]
Hi @JULO01, 
Thanks for your feedback. I'm using Nvim's API to execute the commands by default, but I also experienced that issue and that's why I implemented the `output_pane` option.  Still didn't decide which way to go.

I'm currently using `v0.10.0-dev-2361` and tested both commands. With `output_pane` disabled, none of them are working. With the output_pane enabled:
* `:messages` opens an split view properly with the content
* `:echo 1` doesn't open split window, but `1` is displayed in a notification on the top right (no issue). 

Looks there are some changes between 0.9.5 and 0.10.0, but I can't explain it either. Definitely it's on my list and looking for a solution.

**Comment by juliangoetze** [comment]
Alright, thanks for investigating!

**Comment by jonarrien** [comment]
> But I do have the same issue. I am using `vim-plug` and `NVIM v0.9.5`.

Couldn't figure out the issue with `0.9.5`, please use NVIM `0.10.0` at least.

> * Commands that yield an output with focus on the new window like `:messages` work as expected, opening a new window.
> * For commands that **yield output without focus** like `:echo 1`, the bug remains.

If the output lines are few (< 5), it shows the output on a notification. This can be adjusted in config with: 

```
  output_pane = {
    min_lines = 5,
  }
```

> Do you have any workaround in mind? I would be sad to not use the plugin because of this...

`:echo 1` should show the output in a notification. It works for me at least.
